llvm/include/llvm/Analysis/TargetTransformInfoImpl.h

```diff
@@ -928,8 +904,22 @@ class TargetTransformInfoImplBase {
   virtual InstructionCost
   getMemIntrinsicInstrCost(const MemIntrinsicCostAttributes &MICA,
                            TTI::TargetCostKind CostKind) const {
-    return 1;
+    switch (MICA.getID()) {
+    case Intrinsic::masked_scatter:
+    case Intrinsic::masked_gather:
+    case Intrinsic::masked_load:
+    case Intrinsic::masked_store:
+    case Intrinsic::vp_scatter:
+    case Intrinsic::vp_gather:
+    case Intrinsic::vp_load:
+    case Intrinsic::vp_store:
```

Comment:
Nothing calls `getMemIntrinsicInstrCost` with vp_load/vp_store at the moment right? Should these return an invalid cost by default?

Reply:
Updated. Thanks!